util/easing/src/easing.c

```diff
@@ -69,125 +69,128 @@ static inline float quadratic_in(float step, float max_steps, float max_val)
 {
 	float ratio = step / max_steps;
 
-	return max_val * pow(ratio, 2);
+	return max_val * (ratio * ratio);
 }
 
 static inline float quadratic_out(float step, float max_steps, float max_val)
 {
     float ratio = step / max_steps;
 
-	return -max_val * ratio * (ratio - 2.0);
+	return -max_val * ratio * (ratio - 2.0f);
 }
 
 static inline float quadratic_io(float step, float max_steps, float max_val)
 {
-	float ratio = step / (max_steps / 2.0);
+	float ratio = step / (max_steps / 2.0f);
 
 	if (ratio < 1)
-		return max_val / 2.0 * pow(ratio, 2);
+		return max_val / 2.0f * (ratio * ratio);
 
-    ratio = (step - (max_steps/2.0)) / (max_steps/2.0);
-    return (max_val / 2.0) - (max_val / 2.0) * ratio * (ratio - 2.0);
+    ratio = (step - (max_steps/2.0f)) / (max_steps/2.0f);
+    return (max_val / 2.0f) - (max_val / 2.0f) * ratio * (ratio - 2.0f);
 }
 
 /* Cubic */
 static inline float cubic_in(float step, float max_steps, float max_val)
 {
 	float ratio = step / max_steps;
 
-	return max_val * pow(ratio, 3);
+	return max_val * (ratio * ratio * ratio);
```

Comment:
Whenever you have a `pow` or `sqrt` on the code, your variables are automatically promoted to `double` and the operation is done at a higher precision, then what is done now but simply multiplying `float`s. I am not sure how much this would impact on precision but I assume this is why @mlaz might have asked the previous question about the accuracy.

Reply:
Understood - but I cannot answer that question without a reference. The interface does not suggest that there is `double` precision involved, it's all based on `float` so as a user that's as much as I can expect. In order to answer the question if this refactoring falsifies the result we would need a definition as to what level of deviation is acceptable.

In addition none of the computations uses accumulation which is where `float`'s really start bleeding, all values are monotonically increasing or decreasing and the result is converted to a float. So if the initial value and the result fits into a `float` then every intermediary value will also fit.

Reply:
I believe that at some point, not using the the functions will in fact change the results, we just need to make sure wont break the functions. The method I used to validate these mathematically is to look at the function here and here, then implementing it on wolfram (or some other equivalent tool) run the lib's code in order to produce some values an compare them to what I get on wolfram.